javaagent/sdk-migration-config.yaml

```diff
@@ -0,0 +1,291 @@
+# Copy of https://github.com/open-telemetry/opentelemetry-configuration/blob/v0.3.0/examples/sdk-migration-config.yaml
+# with additional OpenTelemetry Java Agent properties added at .instrumentation.java.
+#
+# sdk-migration-config.yaml is a typical starting point for configuring the SDK when migrating
+# from environment variable based configuration.
```

Comment:
I have mixed feelings about using the migration config in the examples, since it's kind of overwhelming and makes it hard to understand what's relevant to the example and what's not

what do you think of having a minimal yaml for the example, along with warning text at the top about lack of env var support, and pointing people at the migration config if they want/need that?

Reply:
> what do you think of having a minimal yaml for the example, along with warning text at the top about lack of env var support, and pointing people at the migration config if they want/need that?

I have mixed feelings too. The argument to use _one of_ the [opentelemetry-configuration examples](https://github.com/open-telemetry/opentelemetry-configuration/tree/main/examples) as the base is that its less opinionated. As soon as we use a minimal yaml, we have to define what minimal is. If we have a decent definition, maybe we should try to add it to `opentelemetry-configuration`?

One thing I was thinking about with these examples is my experience with configuring database systems like postgres. I'm used to the system being configured with a sensitive default, but when I need to change the default, editing a file like this [postgresql.conf](https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/utils/misc/postgresql.conf.sample) which is a pretty heavily annotated. I like the idea that most of the time you can edit the file based purely on the comments, without having to go lookup some external reference.

Reply:
> One thing I was thinking about with these examples is my experience with configuring database systems like postgres. I'm used to the system being configured with a sensitive default, but when I need to change the default, editing a file like this [postgresql.conf](https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/utils/misc/postgresql.conf.sample) which is a pretty heavily annotated. I like the idea that most of the time you can edit the file based purely on the comments, without having to go lookup some external reference.

this is a good argument

I wonder what to do with the env vars though, since including them in the template makes it seem like those are the defaults

maybe a new template called "-defaults.yaml" that we use outside of migration contexts?

Reply:
We have a template without the env var refs already: https://github.com/open-telemetry/opentelemetry-configuration/blob/main/examples/sdk-config.yaml

Are you suggesting something different?

Reply:
oh, I missed that 🙈

Reply:
Let me go ahead and update this example to be based off `sdk-config.yaml` instead.